app/packages/harbor/src/components/Artifacts.tsx

```diff
@@ -0,0 +1,641 @@
+          noDataTitle="Vulnerabilities not found"
+          refetch={refetch}
+        >
+          <TableContainer>
+            <Table size="small">
+              <TableHead>
+                <TableRow>
+                  <TableCell />
+                  <TableCell>Vulnerability</TableCell>
+                  <TableCell>Severity</TableCell>
+                  <TableCell>Package</TableCell>
+                  <TableCell>Current Version</TableCell>
+                  <TableCell>Fixed in Version</TableCell>
+                </TableRow>
+              </TableHead>
+              <TableBody>
+                {data
+                  ? Object.keys(data).map((key) =>
+                      data[key].vulnerabilities.map((vulnerability) => (
+                        <DetailsVulnerability key={`${key}-${vulnerability.id}`} vulnerability={vulnerability} />
+                      )),
+                    )
+                  : null}
+              </TableBody>
+            </Table>
+          </TableContainer>
+        </UseQueryWrapper>
+      </CardContent>
+    </Card>
+  );
+};
+
+const DetailsBuildHistory: FunctionComponent<{
+  artifactReference: string;
+  instance: IPluginInstance;
+  projectName: string;
+  repositoryName: string;
+}> = ({ instance, projectName, repositoryName, artifactReference }) => {
+  const apiContext = useContext<IAPIContext>(APIContext);
+
+  const { isError, isLoading, error, data, refetch } = useQuery<IBuildHistoryItem[], APIError>(
+    ['harbor/buildhistory', instance, projectName, repositoryName, artifactReference],
+    async () => {
+      return apiContext.client.get<IBuildHistoryItem[]>(
+        `/api/plugins/harbor/buildhistory?projectName=${projectName}&repositoryName=${repositoryName}&artifactReference=${artifactReference}`,
+        {
+          headers: {
+            'x-kobs-cluster': instance.cluster,
+            'x-kobs-plugin': instance.name,
+          },
+        },
+      );
+    },
+  );
+
+  return (
+    <Card sx={{ mb: 6 }}>
+      <CardContent>
+        <Typography variant="h6" pb={2}>
+          Build History
+        </Typography>
+
+        <UseQueryWrapper
+          error={error}
+          errorTitle="Failed to load build history"
+          isError={isError}
+          isLoading={isLoading}
+          isNoData={!data || data.length === 0}
+          noDataTitle="Build history not found"
+          refetch={refetch}
+        >
+          <TableContainer>
+            <Table size="small">
+              <TableHead>
+                <TableRow>
+                  <TableCell>Created on</TableCell>
+                  <TableCell>Command</TableCell>
+                </TableRow>
+              </TableHead>
+              <TableBody>
+                {data?.map((buildHistoryItem, index) => (
+                  <TableRow key={index} sx={{ '&:last-child td, &:last-child th': { border: 0 } }}>
+                    <TableCell sx={{ verticalAlign: 'top', whiteSpace: 'nowrap' }}>
```

Comment:
nit: The "command" column uses up a lot of space for some Dockerfiles (primarly because of very long non-word-breakable text elements). This sometimes causes a horizontal scrollbar on the sidebar.